capreolus/trainer/pytorch.py

```diff
@@ -112,9 +113,12 @@ def single_train_iteration(self, reranker, train_dataloader):
                 self.optimizer.zero_grad()
 
             if (bi + 1) % self.n_batch_per_iter == 0:
-                # REF-TODO: save scheduler state along with optimizer
-                self.lr_scheduler.step()
                 break
+            #     # REF-TODO: save scheduler state along with optimizer
+            #     self.lr_scheduler.step()
+            # hacky: use step instead the internally calculated epoch to support step-wise lr update
+            self.lr_scheduler.step(epoch=cur_step)
```

Comment:
it's a bit hacky here, where by default `lr_scheduler.step` takes in the epoch; changing here as when we passing epoch=0 into our `lr_multiplier` and the warmupiter is also 1, the `lr` would be almost 0 for the entire first epoch.